[Automation] java run fails for com.google.code.gson:gson:2.3

## java run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: java run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.3" ./gradlew clean javaTest -Pcoordinates="com.google.code.gson:gson:2.2.4"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25270673198/job/74092419692
Last 300 lines of the log:
```console
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar
> Task :compileTestJava
> Task :testClasses

> Task :test FAILED

UnsafeAllocatorAnonymous2Test > invokesConfiguredLegacyObjectInputStreamAllocationMethod() FAILED
    java.lang.NoSuchMethodException: com.google.gson.internal.UnsafeAllocator$2.<init>(java.lang.reflect.Method)
        at java.base/java.lang.Class.getConstructor0(Class.java:3763)
        at java.base/java.lang.Class.getDeclaredConstructor(Class.java:2932)
        at com_google_code_gson.gson.UnsafeAllocatorAnonymous2Test.newLegacyObjectInputStreamAllocator(UnsafeAllocatorAnonymous2Test.java:31)
        at com_google_code_gson.gson.UnsafeAllocatorAnonymous2Test.invokesConfiguredLegacyObjectInputStreamAllocationMethod(UnsafeAllocatorAnonymous2Test.java:21)

UnsafeAllocatorAnonymous3Test > invokesConfiguredLegacyObjectStreamClassAllocationMethod() FAILED
    java.lang.NoSuchMethodException: com.google.gson.internal.UnsafeAllocator$3.<init>(java.lang.reflect.Method,int)
        at java.base/java.lang.Class.getConstructor0(Class.java:3763)
        at java.base/java.lang.Class.getDeclaredConstructor(Class.java:2932)
        at com_google_code_gson.gson.UnsafeAllocatorAnonymous3Test.newLegacyObjectStreamClassAllocator(UnsafeAllocatorAnonymous3Test.java:32)
        at com_google_code_gson.gson.UnsafeAllocatorAnonymous3Test.invokesConfiguredLegacyObjectStreamClassAllocationMethod(UnsafeAllocatorAnonymous3Test.java:21)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.code.gson/gson/2.2.4/build/reports/problems/problems-report.html

9 tests completed, 2 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.code.gson/gson/2.2.4/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 7s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
15 actionable tasks: 7 executed, 8 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for com.google.code.gson:gson:2.2.4 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 15s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 2.3 failed with exit code 1.
SOFT-FAIL[native-image run][2.3][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.3" ./gradlew clean test -Pcoordinates="com.google.code.gson:gson:2.2.4"]
========================================================================================
 com.google.code.gson:gson:2.2.4:2.3 stage javac compile
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.code.gson/gson/2.2.4/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 882ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileTestJava

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.code.gson/gson/2.2.4/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1s
10 actionable tasks: 1 executed, 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 3s
11 actionable tasks: 3 executed, 8 up-to-date
========================================================================================
 com.google.code.gson:gson:2.2.4:2.3 stage java run
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.code.gson/gson/2.2.4/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 833ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :classes UP-TO-DATE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
> Task :processTestResources
> Task :compileTestJava
> Task :testClasses

> Task :test FAILED

UnsafeAllocatorAnonymous2Test > invokesConfiguredLegacyObjectInputStreamAllocationMethod() FAILED
    java.lang.NoSuchMethodException: com.google.gson.internal.UnsafeAllocator$2.<init>(java.lang.reflect.Method)
        at java.base/java.lang.Class.getConstructor0(Class.java:3763)
        at java.base/java.lang.Class.getDeclaredConstructor(Class.java:2932)
        at com_google_code_gson.gson.UnsafeAllocatorAnonymous2Test.newLegacyObjectInputStreamAllocator(UnsafeAllocatorAnonymous2Test.java:31)
        at com_google_code_gson.gson.UnsafeAllocatorAnonymous2Test.invokesConfiguredLegacyObjectInputStreamAllocationMethod(UnsafeAllocatorAnonymous2Test.java:21)

UnsafeAllocatorAnonymous3Test > invokesConfiguredLegacyObjectStreamClassAllocationMethod() FAILED
    java.lang.NoSuchMethodException: com.google.gson.internal.UnsafeAllocator$3.<init>(java.lang.reflect.Method,int)
        at java.base/java.lang.Class.getConstructor0(Class.java:3763)
        at java.base/java.lang.Class.getDeclaredConstructor(Class.java:2932)
        at com_google_code_gson.gson.UnsafeAllocatorAnonymous3Test.newLegacyObjectStreamClassAllocator(UnsafeAllocatorAnonymous3Test.java:32)
        at com_google_code_gson.gson.UnsafeAllocatorAnonymous3Test.invokesConfiguredLegacyObjectStreamClassAllocationMethod(UnsafeAllocatorAnonymous3Test.java:21)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.code.gson/gson/2.2.4/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

9 tests completed, 2 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.code.gson/gson/2.2.4/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 2s
14 actionable tasks: 4 executed, 10 up-to-date

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'java run' for 2.3 failed with exit code 1.
FAILED[java run][2.3][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.3" ./gradlew clean javaTest -Pcoordinates="com.google.code.gson:gson:2.2.4"]
```

## java run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: java run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.3" ./gradlew clean javaTest -Pcoordinates="com.google.code.gson:gson:2.2.4"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25270673198/job/74092419655
Last 300 lines of the log:
```console
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar
> Task :compileTestJava
> Task :testClasses

> Task :test FAILED

UnsafeAllocatorAnonymous2Test > invokesConfiguredLegacyObjectInputStreamAllocationMethod() FAILED

9 tests completed, 2 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.code.gson/gson/2.2.4/build/reports/tests/test/index.html
    java.lang.NoSuchMethodException: com.google.gson.internal.UnsafeAllocator$2.<init>(java.lang.reflect.Method)
        at java.base/java.lang.Class.getConstructor0(Class.java:3763)
        at java.base/java.lang.Class.getDeclaredConstructor(Class.java:2932)
        at com_google_code_gson.gson.UnsafeAllocatorAnonymous2Test.newLegacyObjectInputStreamAllocator(UnsafeAllocatorAnonymous2Test.java:31)
        at com_google_code_gson.gson.UnsafeAllocatorAnonymous2Test.invokesConfiguredLegacyObjectInputStreamAllocationMethod(UnsafeAllocatorAnonymous2Test.java:21)

UnsafeAllocatorAnonymous3Test > invokesConfiguredLegacyObjectStreamClassAllocationMethod() FAILED
    java.lang.NoSuchMethodException: com.google.gson.internal.UnsafeAllocator$3.<init>(java.lang.reflect.Method,int)
        at java.base/java.lang.Class.getConstructor0(Class.java:3763)
        at java.base/java.lang.Class.getDeclaredConstructor(Class.java:2932)
        at com_google_code_gson.gson.UnsafeAllocatorAnonymous3Test.newLegacyObjectStreamClassAllocator(UnsafeAllocatorAnonymous3Test.java:32)
        at com_google_code_gson.gson.UnsafeAllocatorAnonymous3Test.invokesConfiguredLegacyObjectStreamClassAllocationMethod(UnsafeAllocatorAnonymous3Test.java:21)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.code.gson/gson/2.2.4/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
15 actionable tasks: 7 executed, 8 up-to-date

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 7s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for com.google.code.gson:gson:2.2.4 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 16s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 2.3 failed with exit code 1.
SOFT-FAIL[native-image run][2.3][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.3" ./gradlew clean test -Pcoordinates="com.google.code.gson:gson:2.2.4"]
========================================================================================
 com.google.code.gson:gson:2.2.4:2.3 stage javac compile
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.code.gson/gson/2.2.4/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 957ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileTestJava

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.code.gson/gson/2.2.4/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1s
10 actionable tasks: 1 executed, 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 3s
11 actionable tasks: 3 executed, 8 up-to-date
========================================================================================
 com.google.code.gson:gson:2.2.4:2.3 stage java run
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.code.gson/gson/2.2.4/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 892ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :tck-test-support:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava
> Task :processTestResources
> Task :testClasses

> Task :test FAILED

UnsafeAllocatorAnonymous2Test > invokesConfiguredLegacyObjectInputStreamAllocationMethod() FAILED
    java.lang.NoSuchMethodException: com.google.gson.internal.UnsafeAllocator$2.<init>(java.lang.reflect.Method)
        at java.base/java.lang.Class.getConstructor0(Class.java:3763)
        at java.base/java.lang.Class.getDeclaredConstructor(Class.java:2932)
        at com_google_code_gson.gson.UnsafeAllocatorAnonymous2Test.newLegacyObjectInputStreamAllocator(UnsafeAllocatorAnonymous2Test.java:31)
        at com_google_code_gson.gson.UnsafeAllocatorAnonymous2Test.invokesConfiguredLegacyObjectInputStreamAllocationMethod(UnsafeAllocatorAnonymous2Test.java:21)

UnsafeAllocatorAnonymous3Test > invokesConfiguredLegacyObjectStreamClassAllocationMethod() FAILED
    java.lang.NoSuchMethodException: com.google.gson.internal.UnsafeAllocator$3.<init>(java.lang.reflect.Method,int)
        at java.base/java.lang.Class.getConstructor0(Class.java:3763)
        at java.base/java.lang.Class.getDeclaredConstructor(Class.java:2932)
        at com_google_code_gson.gson.UnsafeAllocatorAnonymous3Test.newLegacyObjectStreamClassAllocator(UnsafeAllocatorAnonymous3Test.java:32)
        at com_google_code_gson.gson.UnsafeAllocatorAnonymous3Test.invokesConfiguredLegacyObjectStreamClassAllocationMethod(UnsafeAllocatorAnonymous3Test.java:21)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.code.gson/gson/2.2.4/build/reports/problems/problems-report.html

9 tests completed, 2 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.code.gson/gson/2.2.4/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
 in 2s
14 actionable tasks: 4 executed, 10 up-to-date

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'java run' for 2.3 failed with exit code 1.
FAILED[java run][2.3][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.3" ./gradlew clean javaTest -Pcoordinates="com.google.code.gson:gson:2.2.4"]
```

## java run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: java run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="2.3" ./gradlew clean javaTest -Pcoordinates="com.google.code.gson:gson:2.2.4"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25270673198/job/74092419629
Last 300 lines of the log:
```console
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar
> Task :compileTestJava
> Task :testClasses

> Task :test FAILED

UnsafeAllocatorAnonymous2Test > invokesConfiguredLegacyObjectInputStreamAllocationMethod() FAILED
    java.lang.NoSuchMethodException: com.google.gson.internal.UnsafeAllocator$2.<init>(java.lang.reflect.Method)
        at java.base/java.lang.Class.getConstructor0(Class.java:3763)
        at java.base/java.lang.Class.getDeclaredConstructor(Class.java:2932)
        at com_google_code_gson.gson.UnsafeAllocatorAnonymous2Test.newLegacyObjectInputStreamAllocator(UnsafeAllocatorAnonymous2Test.java:31)
        at com_google_code_gson.gson.UnsafeAllocatorAnonymous2Test.invokesConfiguredLegacyObjectInputStreamAllocationMethod(UnsafeAllocatorAnonymous2Test.java:21)

UnsafeAllocatorAnonymous3Test > invokesConfiguredLegacyObjectStreamClassAllocationMethod() FAILED
    java.lang.NoSuchMethodException: com.google.gson.internal.UnsafeAllocator$3.<init>(java.lang.reflect.Method,int)
        at java.base/java.lang.Class.getConstructor0(Class.java:3763)
        at java.base/java.lang.Class.getDeclaredConstructor(Class.java:2932)
        at com_google_code_gson.gson.UnsafeAllocatorAnonymous3Test.newLegacyObjectStreamClassAllocator(UnsafeAllocatorAnonymous3Test.java:32)
        at com_google_code_gson.gson.UnsafeAllocatorAnonymous3Test.invokesConfiguredLegacyObjectStreamClassAllocationMethod(UnsafeAllocatorAnonymous3Test.java:21)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.code.gson/gson/2.2.4/build/reports/problems/problems-report.html

9 tests completed, 2 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.code.gson/gson/2.2.4/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 7s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
15 actionable tasks: 7 executed, 8 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for com.google.code.gson:gson:2.2.4 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 15s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 2.3 failed with exit code 1.
SOFT-FAIL[native-image run][2.3][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="2.3" ./gradlew clean test -Pcoordinates="com.google.code.gson:gson:2.2.4"]
========================================================================================
 com.google.code.gson:gson:2.2.4:2.3 stage javac compile
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.code.gson/gson/2.2.4/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 951ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileTestJava

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.code.gson/gson/2.2.4/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1s
10 actionable tasks: 1 executed, 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 3s
11 actionable tasks: 3 executed, 8 up-to-date
========================================================================================
 com.google.code.gson:gson:2.2.4:2.3 stage java run
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.code.gson/gson/2.2.4/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 831ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :tck-test-support:jar UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava
> Task :processTestResources
> Task :testClasses

> Task :test FAILED

UnsafeAllocatorAnonymous2Test > invokesConfiguredLegacyObjectInputStreamAllocationMethod() FAILED
    java.lang.NoSuchMethodException: com.google.gson.internal.UnsafeAllocator$2.<init>(java.lang.reflect.Method)
        at java.base/java.lang.Class.getConstructor0(Class.java:3763)
        at java.base/java.lang.Class.getDeclaredConstructor(Class.java:2932)
        at com_google_code_gson.gson.UnsafeAllocatorAnonymous2Test.newLegacyObjectInputStreamAllocator(UnsafeAllocatorAnonymous2Test.java:31)
        at com_google_code_gson.gson.UnsafeAllocatorAnonymous2Test.invokesConfiguredLegacyObjectInputStreamAllocationMethod(UnsafeAllocatorAnonymous2Test.java:21)

UnsafeAllocatorAnonymous3Test > invokesConfiguredLegacyObjectStreamClassAllocationMethod() FAILED
    java.lang.NoSuchMethodException: com.google.gson.internal.UnsafeAllocator$3.<init>(java.lang.reflect.Method,int)
        at java.base/java.lang.Class.getConstructor0(Class.java:3763)
        at java.base/java.lang.Class.getDeclaredConstructor(Class.java:2932)
        at com_google_code_gson.gson.UnsafeAllocatorAnonymous3Test.newLegacyObjectStreamClassAllocator(UnsafeAllocatorAnonymous3Test.java:32)
        at com_google_code_gson.gson.UnsafeAllocatorAnonymous3Test.invokesConfiguredLegacyObjectStreamClassAllocationMethod(UnsafeAllocatorAnonymous3Test.java:21)
9 tests completed, 2 failed


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.code.gson/gson/2.2.4/build/reports/problems/problems-report.html
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.google.code.gson/gson/2.2.4/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).


Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
14 actionable tasks: 4 executed, 10 up-to-date

BUILD FAILED in 2s

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'java run' for 2.3 failed with exit code 1.
FAILED[java run][2.3][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="2.3" ./gradlew clean javaTest -Pcoordinates="com.google.code.gson:gson:2.2.4"]
```

--- a/metadata/com.google.code.gson/gson/2.3/reachability-metadata.json
+++ b/metadata/com.google.code.gson/gson/2.3/reachability-metadata.json
@@ -1,0 +1,29 @@
+{
+  "reflection" : [
+    {
+      "condition" : {
+        "typeReached" : "com.google.gson.internal.UnsafeAllocator"
+      },
+      "type" : "sun.misc.Unsafe",
+      "fields" : [
+        {
+          "name" : "theUnsafe"
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.google.gson.internal.UnsafeAllocator$1"
+      },
+      "type" : "sun.misc.Unsafe",
+      "methods" : [
+        {
+          "name" : "allocateInstance",
+          "parameterTypes" : [
+            "java.lang.Class"
+          ]
+        }
+      ]
+    }
+  ]
+}

--- a/metadata/com.google.code.gson/gson/index.json
+++ b/metadata/com.google.code.gson/gson/index.json
@@ -2,6 +2,11 @@
   {
     "latest" : true,
     "metadata-version" : "2.3",
+    "source-code-url" : "https://repo1.maven.org/maven2/com/google/code/gson/gson/$version$/gson-$version$-sources.jar",
+    "repository-url" : "https://github.com/google/gson",
+    "test-code-url" : "https://github.com/google/gson/tree/gson-$version$/src/test",
+    "documentation-url" : "https://repo1.maven.org/maven2/com/google/code/gson/gson/$version$/gson-$version$-javadoc.jar",
+    "description" : "Gson is a Java library for converting Java objects to and from JSON. It provides object mapping, tree, and streaming APIs for JSON serialization and deserialization.",
     "tested-versions" : [
       "2.3"
     ],

--- a/metadata/com.google.code.gson/gson/index.json
+++ b/metadata/com.google.code.gson/gson/index.json
@@ -1,6 +1,15 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "2.3",
+    "tested-versions" : [
+      "2.3"
+    ],
+    "allowed-packages" : [
+      "com.google.gson"
+    ]
+  },
+  {
     "metadata-version" : "2.2.4",
     "source-code-url" : "https://repo1.maven.org/maven2/com/google/code/gson/gson/$version$/gson-$version$-sources.jar",
     "repository-url" : "https://github.com/google/gson",

--- a/stats/com.google.code.gson/gson/2.3/execution-metrics.json
+++ b/stats/com.google.code.gson/gson/2.3/execution-metrics.json
@@ -1,0 +1,101 @@
+{
+  "fix_java_run_fail:2026-06-06": {
+    "timestamp": "2026-06-06T18:56:26.092590Z",
+    "library": "com.google.code.gson:gson:2.3",
+    "starting_commit": "c8f60e3d19d441a0bad9f73b9916b549b102f0e6",
+    "ending_commit": "ce67e4ee78c6ce607ddd97a4123b64406f3afea3",
+    "previous_library": "com.google.code.gson:gson:2.2.4",
+    "strategy_name": "java_run_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "2.3",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 13,
+            "totalCalls": 13
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 13,
+        "totalCalls": 13
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 4027,
+          "missed": 13364,
+          "ratio": 0.231557,
+          "total": 17391
+        },
+        "line": {
+          "covered": 953,
+          "missed": 2976,
+          "ratio": 0.242555,
+          "total": 3929
+        },
+        "method": {
+          "covered": 233,
+          "missed": 573,
+          "ratio": 0.289082,
+          "total": 806
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "2.2.4",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 13,
+            "totalCalls": 13
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 13,
+        "totalCalls": 13
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 3910,
+          "missed": 11376,
+          "ratio": 0.25579,
+          "total": 15286
+        },
+        "line": {
+          "covered": 906,
+          "missed": 2481,
+          "ratio": 0.267493,
+          "total": 3387
+        },
+        "method": {
+          "covered": 230,
+          "missed": 503,
+          "ratio": 0.313779,
+          "total": 733
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 23548,
+      "cached_input_tokens_used": 126976,
+      "output_tokens_used": 2714,
+      "iterations": 1,
+      "cost_usd": 0.1449,
+      "tested_library_loc": 953,
+      "metadata_entries": 4,
+      "test_only_metadata_entries": 22,
+      "previous_library_metadata_entries": 4,
+      "previous_library_test_only_metadata_entries": 20,
+      "code_coverage_percent": 24.26,
+      "previous_library_coverage_percent": 26.75
+    },
+    "artifacts": {
+      "test_file": "tests/src/com.google.code.gson/gson/2.3/src/test/java/com_google_code_gson/gson/ArrayTypeAdapterTest.java",
+      "metadata_file": "metadata/com.google.code.gson/gson/2.3/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/com.google.code.gson/gson/2.3/stats.json
+++ b/stats/com.google.code.gson/gson/2.3/stats.json
@@ -1,0 +1,37 @@
+{
+  "versions" : [ {
+    "version" : "2.3",
+    "dynamicAccess" : {
+      "breakdown" : {
+        "reflection" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 13,
+          "totalCalls" : 13
+        }
+      },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 13,
+      "totalCalls" : 13
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 4027,
+        "missed" : 13364,
+        "ratio" : 0.231557,
+        "total" : 17391
+      },
+      "line" : {
+        "covered" : 953,
+        "missed" : 2976,
+        "ratio" : 0.242555,
+        "total" : 3929
+      },
+      "method" : {
+        "covered" : 233,
+        "missed" : 573,
+        "ratio" : 0.289082,
+        "total" : 806
+      }
+    }
+  } ]
+}

--- a/tests/src/com.google.code.gson/gson/2.3/.gitignore
+++ b/tests/src/com.google.code.gson/gson/2.3/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/com.google.code.gson/gson/2.3/build.gradle
+++ b/tests/src/com.google.code.gson/gson/2.3/build.gradle
@@ -1,0 +1,27 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "com.google.code.gson:gson:$libraryVersion"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/com.google.code.gson/gson/2.3/gradle.properties
+++ b/tests/src/com.google.code.gson/gson/2.3/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = com.google.code.gson:gson:2.3
+library.version = 2.3
+metadata.dir = com.google.code.gson/gson/2.3/

--- a/tests/src/com.google.code.gson/gson/2.3/settings.gradle
+++ b/tests/src/com.google.code.gson/gson/2.3/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'com.google.code.gson.gson_tests'

--- a/tests/src/com.google.code.gson/gson/2.3/src/test/java/com_google_code_gson/gson/ArrayTypeAdapterTest.java
+++ b/tests/src/com.google.code.gson/gson/2.3/src/test/java/com_google_code_gson/gson/ArrayTypeAdapterTest.java
@@ -1,0 +1,45 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package com_google_code_gson.gson;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.gson.Gson;
+import org.junit.jupiter.api.Test;
+
+public class ArrayTypeAdapterTest {
+    private final Gson gson = new Gson();
+
+    @Test
+    void deserializesObjectArrayThroughReflectiveArrayAllocation() {
+        String json = """
+                [
+                  {"name":"paperclips","quantity":3,"available":true},
+                  {"name":"folders","quantity":9,"available":false}
+                ]
+                """;
+
+        InventoryItem[] items = gson.fromJson(json, InventoryItem[].class);
+
+        assertThat(items).hasSize(2);
+        assertThat(items[0].name).isEqualTo("paperclips");
+        assertThat(items[0].quantity).isEqualTo(3);
+        assertThat(items[0].available).isTrue();
+        assertThat(items[1].name).isEqualTo("folders");
+        assertThat(items[1].quantity).isEqualTo(9);
+        assertThat(items[1].available).isFalse();
+    }
+
+    public static final class InventoryItem {
+        private String name;
+        private int quantity;
+        private boolean available;
+
+        public InventoryItem() {
+        }
+    }
+}

--- a/tests/src/com.google.code.gson/gson/2.3/src/test/java/com_google_code_gson/gson/EnumTypeAdapterTest.java
+++ b/tests/src/com.google.code.gson/gson/2.3/src/test/java/com_google_code_gson/gson/EnumTypeAdapterTest.java
@@ -1,0 +1,32 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package com_google_code_gson.gson;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.gson.Gson;
+import com.google.gson.annotations.SerializedName;
+import org.junit.jupiter.api.Test;
+
+public class EnumTypeAdapterTest {
+    private final Gson gson = new Gson();
+
+    @Test
+    void resolvesEnumConstantsThroughReflectedPublicFields() {
+        ShipmentState state = gson.fromJson("\"shipped\"", ShipmentState.class);
+
+        assertThat(state).isEqualTo(ShipmentState.IN_TRANSIT);
+        assertThat(gson.toJson(ShipmentState.READY)).isEqualTo("\"ready\"");
+    }
+
+    public enum ShipmentState {
+        @SerializedName("ready")
+        READY,
+        @SerializedName("shipped")
+        IN_TRANSIT
+    }
+}

--- a/tests/src/com.google.code.gson/gson/2.3/src/test/java/com_google_code_gson/gson/FieldAttributesTest.java
+++ b/tests/src/com.google.code.gson/gson/2.3/src/test/java/com_google_code_gson/gson/FieldAttributesTest.java
@@ -1,0 +1,40 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package com_google_code_gson.gson;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.gson.FieldAttributes;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import org.junit.jupiter.api.Test;
+
+public class FieldAttributesTest {
+    @Test
+    void readsValueFromWrappedPublicField() throws Exception {
+        Field field = FieldCarrier.class.getField("title");
+        FieldAttributes attributes = new FieldAttributes(field);
+        FieldCarrier carrier = new FieldCarrier("integration-guide");
+
+        Method getMethod = FieldAttributes.class.getDeclaredMethod("get", Object.class);
+        getMethod.setAccessible(true);
+        Object value = getMethod.invoke(attributes, carrier);
+
+        assertThat(value).isEqualTo("integration-guide");
+        assertThat(attributes.getName()).isEqualTo("title");
+        assertThat(attributes.getDeclaredClass()).isEqualTo(String.class);
+        assertThat(attributes.getDeclaringClass()).isEqualTo(FieldCarrier.class);
+    }
+
+    public static final class FieldCarrier {
+        public String title;
+
+        public FieldCarrier(String title) {
+            this.title = title;
+        }
+    }
+}

--- a/tests/src/com.google.code.gson/gson/2.3/src/test/java/com_google_code_gson/gson/InnerGsonInnerTypesTest.java
+++ b/tests/src/com.google.code.gson/gson/2.3/src/test/java/com_google_code_gson/gson/InnerGsonInnerTypesTest.java
@@ -1,0 +1,32 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package com_google_code_gson.gson;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.gson.internal.$Gson$Types;
+import java.lang.reflect.GenericArrayType;
+import java.lang.reflect.ParameterizedType;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+
+public class InnerGsonInnerTypesTest {
+    @Test
+    void resolvesRawClassForGenericArrayType() {
+        ParameterizedType listOfStrings = $Gson$Types.newParameterizedTypeWithOwner(
+                null,
+                List.class,
+                String.class);
+        GenericArrayType arrayOfLists = $Gson$Types.arrayOf(listOfStrings);
+
+        Class<?> rawType = $Gson$Types.getRawType(arrayOfLists);
+
+        assertThat(rawType).isEqualTo(List[].class);
+        assertThat(rawType.isArray()).isTrue();
+        assertThat(rawType.getComponentType()).isEqualTo(List.class);
+    }
+}

--- a/tests/src/com.google.code.gson/gson/2.3/src/test/java/com_google_code_gson/gson/ReflectiveTypeAdapterFactoryAnonymous1Test.java
+++ b/tests/src/com.google.code.gson/gson/2.3/src/test/java/com_google_code_gson/gson/ReflectiveTypeAdapterFactoryAnonymous1Test.java
@@ -1,0 +1,53 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package com_google_code_gson.gson;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.gson.Gson;
+import org.junit.jupiter.api.Test;
+
+public class ReflectiveTypeAdapterFactoryAnonymous1Test {
+    private final Gson gson = new Gson();
+
+    @Test
+    void serializesObjectFieldsThroughReflectiveBoundFields() {
+        InventoryItem item = new InventoryItem("desk-lamp", 12, true);
+
+        String json = gson.toJson(item);
+
+        assertThat(json).contains("\"name\":\"desk-lamp\"");
+        assertThat(json).contains("\"quantity\":12");
+        assertThat(json).contains("\"available\":true");
+    }
+
+    @Test
+    void deserializesObjectFieldsThroughReflectiveBoundFields() {
+        String json = "{\"name\":\"notebook\",\"quantity\":5,\"available\":false}";
+
+        InventoryItem item = gson.fromJson(json, InventoryItem.class);
+
+        assertThat(item.name).isEqualTo("notebook");
+        assertThat(item.quantity).isEqualTo(5);
+        assertThat(item.available).isFalse();
+    }
+
+    public static final class InventoryItem {
+        private String name;
+        private int quantity;
+        private boolean available;
+
+        public InventoryItem() {
+        }
+
+        InventoryItem(String name, int quantity, boolean available) {
+            this.name = name;
+            this.quantity = quantity;
+            this.available = available;
+        }
+    }
+}

--- a/tests/src/com.google.code.gson/gson/2.3/src/test/java/com_google_code_gson/gson/UnsafeAllocatorAnonymous2Test.java
+++ b/tests/src/com.google.code.gson/gson/2.3/src/test/java/com_google_code_gson/gson/UnsafeAllocatorAnonymous2Test.java
@@ -1,0 +1,49 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package com_google_code_gson.gson;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.gson.internal.UnsafeAllocator;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import org.junit.jupiter.api.Test;
+
+public class UnsafeAllocatorAnonymous2Test {
+    @Test
+    void invokesConfiguredLegacyObjectInputStreamAllocationMethod() throws Exception {
+        Method fallbackMethod = UnsafeAllocatorAnonymous2Test.class.getDeclaredMethod(
+                "legacyObjectInputStreamNewInstance", Class.class, Class.class);
+        UnsafeAllocator allocator = newLegacyObjectInputStreamAllocator(fallbackMethod);
+
+        FallbackAllocatedMessage message = allocator.newInstance(FallbackAllocatedMessage.class);
+
+        assertThat(message.instantiationClass).isEqualTo(FallbackAllocatedMessage.class);
+        assertThat(message.constructorClass).isEqualTo(Object.class);
+    }
+
+    private static UnsafeAllocator newLegacyObjectInputStreamAllocator(Method fallbackMethod) throws Exception {
+        Class<?> allocatorClass = Class.forName("com.google.gson.internal.UnsafeAllocator$2");
+        Constructor<?> constructor = allocatorClass.getDeclaredConstructor(Method.class);
+        constructor.setAccessible(true);
+        return (UnsafeAllocator) constructor.newInstance(fallbackMethod);
+    }
+
+    public static Object legacyObjectInputStreamNewInstance(Class<?> instantiationClass, Class<?> constructorClass) {
+        return new FallbackAllocatedMessage(instantiationClass, constructorClass);
+    }
+
+    public static final class FallbackAllocatedMessage {
+        private final Class<?> instantiationClass;
+        private final Class<?> constructorClass;
+
+        FallbackAllocatedMessage(Class<?> instantiationClass, Class<?> constructorClass) {
+            this.instantiationClass = instantiationClass;
+            this.constructorClass = constructorClass;
+        }
+    }
+}

--- a/tests/src/com.google.code.gson/gson/2.3/src/test/java/com_google_code_gson/gson/UnsafeAllocatorAnonymous2Test.java
+++ b/tests/src/com.google.code.gson/gson/2.3/src/test/java/com_google_code_gson/gson/UnsafeAllocatorAnonymous2Test.java
@@ -15,35 +15,36 @@ import org.junit.jupiter.api.Test;
 
 public class UnsafeAllocatorAnonymous2Test {
     @Test
-    void invokesConfiguredLegacyObjectInputStreamAllocationMethod() throws Exception {
+    void invokesConfiguredLegacyObjectStreamClassAllocationMethod() throws Exception {
         Method fallbackMethod = UnsafeAllocatorAnonymous2Test.class.getDeclaredMethod(
-                "legacyObjectInputStreamNewInstance", Class.class, Class.class);
-        UnsafeAllocator allocator = newLegacyObjectInputStreamAllocator(fallbackMethod);
+                "legacyObjectStreamClassNewInstance", Class.class, int.class);
+        UnsafeAllocator allocator = newLegacyObjectStreamClassAllocator(fallbackMethod, 41);
 
         FallbackAllocatedMessage message = allocator.newInstance(FallbackAllocatedMessage.class);
 
         assertThat(message.instantiationClass).isEqualTo(FallbackAllocatedMessage.class);
-        assertThat(message.constructorClass).isEqualTo(Object.class);
+        assertThat(message.constructorId).isEqualTo(41);
     }
 
-    private static UnsafeAllocator newLegacyObjectInputStreamAllocator(Method fallbackMethod) throws Exception {
+    private static UnsafeAllocator newLegacyObjectStreamClassAllocator(Method fallbackMethod, int constructorId)
+            throws Exception {
         Class<?> allocatorClass = Class.forName("com.google.gson.internal.UnsafeAllocator$2");
-        Constructor<?> constructor = allocatorClass.getDeclaredConstructor(Method.class);
+        Constructor<?> constructor = allocatorClass.getDeclaredConstructor(Method.class, int.class);
         constructor.setAccessible(true);
-        return (UnsafeAllocator) constructor.newInstance(fallbackMethod);
+        return (UnsafeAllocator) constructor.newInstance(fallbackMethod, constructorId);
     }
 
-    public static Object legacyObjectInputStreamNewInstance(Class<?> instantiationClass, Class<?> constructorClass) {
-        return new FallbackAllocatedMessage(instantiationClass, constructorClass);
+    public static Object legacyObjectStreamClassNewInstance(Class<?> instantiationClass, int constructorId) {
+        return new FallbackAllocatedMessage(instantiationClass, constructorId);
     }
 
     public static final class FallbackAllocatedMessage {
         private final Class<?> instantiationClass;
-        private final Class<?> constructorClass;
+        private final int constructorId;
 
-        FallbackAllocatedMessage(Class<?> instantiationClass, Class<?> constructorClass) {
+        FallbackAllocatedMessage(Class<?> instantiationClass, int constructorId) {
             this.instantiationClass = instantiationClass;
-            this.constructorClass = constructorClass;
+            this.constructorId = constructorId;
         }
     }
 }

--- a/tests/src/com.google.code.gson/gson/2.3/src/test/java/com_google_code_gson/gson/UnsafeAllocatorAnonymous3Test.java
+++ b/tests/src/com.google.code.gson/gson/2.3/src/test/java/com_google_code_gson/gson/UnsafeAllocatorAnonymous3Test.java
@@ -1,0 +1,50 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package com_google_code_gson.gson;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.gson.internal.UnsafeAllocator;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import org.junit.jupiter.api.Test;
+
+public class UnsafeAllocatorAnonymous3Test {
+    @Test
+    void invokesConfiguredLegacyObjectStreamClassAllocationMethod() throws Exception {
+        Method fallbackMethod = UnsafeAllocatorAnonymous3Test.class.getDeclaredMethod(
+                "legacyObjectStreamClassNewInstance", Class.class, int.class);
+        UnsafeAllocator allocator = newLegacyObjectStreamClassAllocator(fallbackMethod, 41);
+
+        FallbackAllocatedMessage message = allocator.newInstance(FallbackAllocatedMessage.class);
+
+        assertThat(message.instantiationClass).isEqualTo(FallbackAllocatedMessage.class);
+        assertThat(message.constructorId).isEqualTo(41);
+    }
+
+    private static UnsafeAllocator newLegacyObjectStreamClassAllocator(Method fallbackMethod, int constructorId)
+            throws Exception {
+        Class<?> allocatorClass = Class.forName("com.google.gson.internal.UnsafeAllocator$3");
+        Constructor<?> constructor = allocatorClass.getDeclaredConstructor(Method.class, int.class);
+        constructor.setAccessible(true);
+        return (UnsafeAllocator) constructor.newInstance(fallbackMethod, constructorId);
+    }
+
+    public static Object legacyObjectStreamClassNewInstance(Class<?> instantiationClass, int constructorId) {
+        return new FallbackAllocatedMessage(instantiationClass, constructorId);
+    }
+
+    public static final class FallbackAllocatedMessage {
+        private final Class<?> instantiationClass;
+        private final int constructorId;
+
+        FallbackAllocatedMessage(Class<?> instantiationClass, int constructorId) {
+            this.instantiationClass = instantiationClass;
+            this.constructorId = constructorId;
+        }
+    }
+}

--- a/tests/src/com.google.code.gson/gson/2.3/src/test/java/com_google_code_gson/gson/UnsafeAllocatorAnonymous3Test.java
+++ b/tests/src/com.google.code.gson/gson/2.3/src/test/java/com_google_code_gson/gson/UnsafeAllocatorAnonymous3Test.java
@@ -15,36 +15,35 @@ import org.junit.jupiter.api.Test;
 
 public class UnsafeAllocatorAnonymous3Test {
     @Test
-    void invokesConfiguredLegacyObjectStreamClassAllocationMethod() throws Exception {
+    void invokesConfiguredLegacyObjectInputStreamAllocationMethod() throws Exception {
         Method fallbackMethod = UnsafeAllocatorAnonymous3Test.class.getDeclaredMethod(
-                "legacyObjectStreamClassNewInstance", Class.class, int.class);
-        UnsafeAllocator allocator = newLegacyObjectStreamClassAllocator(fallbackMethod, 41);
+                "legacyObjectInputStreamNewInstance", Class.class, Class.class);
+        UnsafeAllocator allocator = newLegacyObjectInputStreamAllocator(fallbackMethod);
 
         FallbackAllocatedMessage message = allocator.newInstance(FallbackAllocatedMessage.class);
 
         assertThat(message.instantiationClass).isEqualTo(FallbackAllocatedMessage.class);
-        assertThat(message.constructorId).isEqualTo(41);
+        assertThat(message.constructorClass).isEqualTo(Object.class);
     }
 
-    private static UnsafeAllocator newLegacyObjectStreamClassAllocator(Method fallbackMethod, int constructorId)
-            throws Exception {
+    private static UnsafeAllocator newLegacyObjectInputStreamAllocator(Method fallbackMethod) throws Exception {
         Class<?> allocatorClass = Class.forName("com.google.gson.internal.UnsafeAllocator$3");
-        Constructor<?> constructor = allocatorClass.getDeclaredConstructor(Method.class, int.class);
+        Constructor<?> constructor = allocatorClass.getDeclaredConstructor(Method.class);
         constructor.setAccessible(true);
-        return (UnsafeAllocator) constructor.newInstance(fallbackMethod, constructorId);
+        return (UnsafeAllocator) constructor.newInstance(fallbackMethod);
     }
 
-    public static Object legacyObjectStreamClassNewInstance(Class<?> instantiationClass, int constructorId) {
-        return new FallbackAllocatedMessage(instantiationClass, constructorId);
+    public static Object legacyObjectInputStreamNewInstance(Class<?> instantiationClass, Class<?> constructorClass) {
+        return new FallbackAllocatedMessage(instantiationClass, constructorClass);
     }
 
     public static final class FallbackAllocatedMessage {
         private final Class<?> instantiationClass;
-        private final int constructorId;
+        private final Class<?> constructorClass;
 
-        FallbackAllocatedMessage(Class<?> instantiationClass, int constructorId) {
+        FallbackAllocatedMessage(Class<?> instantiationClass, Class<?> constructorClass) {
             this.instantiationClass = instantiationClass;
-            this.constructorId = constructorId;
+            this.constructorClass = constructorClass;
         }
     }
 }

--- a/tests/src/com.google.code.gson/gson/2.3/src/test/java/com_google_code_gson/gson/UnsafeAllocatorTest.java
+++ b/tests/src/com.google.code.gson/gson/2.3/src/test/java/com_google_code_gson/gson/UnsafeAllocatorTest.java
@@ -1,0 +1,43 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package com_google_code_gson.gson;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.gson.Gson;
+import org.junit.jupiter.api.Test;
+
+public class UnsafeAllocatorTest {
+    private final Gson gson = new Gson();
+
+    @Test
+    void deserializesTypeWithoutNoArgsConstructorUsingUnsafeAllocation() {
+        ConstructorOnlyMessage message = gson.fromJson(
+                "{\"label\":\"created-from-json\",\"priority\":7}",
+                ConstructorOnlyMessage.class);
+
+        assertThat(message.label).isEqualTo("created-from-json");
+        assertThat(message.priority).isEqualTo(7);
+        assertThat(message.wasConstructorInvoked()).isFalse();
+    }
+
+    public static final class ConstructorOnlyMessage {
+        private final transient boolean constructorInvoked;
+        private String label;
+        private int priority;
+
+        public ConstructorOnlyMessage(String label, int priority) {
+            this.constructorInvoked = true;
+            this.label = label;
+            this.priority = priority;
+        }
+
+        boolean wasConstructorInvoked() {
+            return constructorInvoked;
+        }
+    }
+}

--- a/tests/src/com.google.code.gson/gson/2.3/src/test/resources/META-INF/native-image/reachability-metadata.json
+++ b/tests/src/com.google.code.gson/gson/2.3/src/test/resources/META-INF/native-image/reachability-metadata.json
@@ -1,0 +1,115 @@
+{
+  "reflection" : [
+    {
+      "condition" : {
+        "typeReached" : "com.google.gson.internal.bind.ArrayTypeAdapter$1"
+      },
+      "type" : "com_google_code_gson.gson.ArrayTypeAdapterTest$InventoryItem"
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper"
+      },
+      "type" : "com_google_code_gson.gson.ArrayTypeAdapterTest$InventoryItem",
+      "fields" : [
+        {
+          "name" : "available"
+        },
+        {
+          "name" : "name"
+        },
+        {
+          "name" : "quantity"
+        }
+      ],
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.google.gson.internal.bind.ArrayTypeAdapter"
+      },
+      "type" : "com_google_code_gson.gson.ArrayTypeAdapterTest$InventoryItem[]"
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.google.gson.reflect.TypeToken"
+      },
+      "type" : "com_google_code_gson.gson.ArrayTypeAdapterTest$InventoryItem[]"
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.google.gson.internal.bind.TypeAdapters$EnumTypeAdapter"
+      },
+      "type" : "com_google_code_gson.gson.EnumTypeAdapterTest$ShipmentState"
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.google.gson.FieldAttributes"
+      },
+      "type" : "com_google_code_gson.gson.FieldAttributesTest$FieldCarrier",
+      "fields" : [
+        {
+          "name" : "title"
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.google.gson.Gson"
+      },
+      "type" : "com_google_code_gson.gson.ReflectiveTypeAdapterFactoryAnonymous1Test$InventoryItem"
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1"
+      },
+      "type" : "com_google_code_gson.gson.ReflectiveTypeAdapterFactoryAnonymous1Test$InventoryItem",
+      "fields" : [
+        {
+          "name" : "available"
+        },
+        {
+          "name" : "name"
+        },
+        {
+          "name" : "quantity"
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.google.gson.internal.UnsafeAllocator$2"
+      },
+      "type" : "com_google_code_gson.gson.UnsafeAllocatorAnonymous2Test",
+      "methods" : [
+        {
+          "name" : "legacyObjectInputStreamNewInstance",
+          "parameterTypes" : [
+            "java.lang.Class",
+            "java.lang.Class"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "com.google.gson.internal.UnsafeAllocator$3"
+      },
+      "type" : "com_google_code_gson.gson.UnsafeAllocatorAnonymous3Test",
+      "methods" : [
+        {
+          "name" : "legacyObjectStreamClassNewInstance",
+          "parameterTypes" : [
+            "java.lang.Class",
+            "int"
+          ]
+        }
+      ]
+    }
+  ]
+}

--- a/tests/src/com.google.code.gson/gson/2.3/src/test/resources/META-INF/native-image/reachability-metadata.json
+++ b/tests/src/com.google.code.gson/gson/2.3/src/test/resources/META-INF/native-image/reachability-metadata.json
@@ -83,15 +83,22 @@
     },
     {
       "condition" : {
+        "typeReached" : "com.google.gson.internal.UnsafeAllocator$1"
+      },
+      "type" : "com_google_code_gson.gson.UnsafeAllocatorTest$ConstructorOnlyMessage",
+      "unsafeAllocated" : true
+    },
+    {
+      "condition" : {
         "typeReached" : "com.google.gson.internal.UnsafeAllocator$2"
       },
       "type" : "com_google_code_gson.gson.UnsafeAllocatorAnonymous2Test",
       "methods" : [
         {
-          "name" : "legacyObjectInputStreamNewInstance",
+          "name" : "legacyObjectStreamClassNewInstance",
           "parameterTypes" : [
             "java.lang.Class",
-            "java.lang.Class"
+            "int"
           ]
         }
       ]
@@ -103,10 +110,10 @@
       "type" : "com_google_code_gson.gson.UnsafeAllocatorAnonymous3Test",
       "methods" : [
         {
-          "name" : "legacyObjectStreamClassNewInstance",
+          "name" : "legacyObjectInputStreamNewInstance",
           "parameterTypes" : [
             "java.lang.Class",
-            "int"
+            "java.lang.Class"
           ]
         }
       ]

--- a/tests/src/com.google.code.gson/gson/2.3/user-code-filter.json
+++ b/tests/src/com.google.code.gson/gson/2.3/user-code-filter.json
@@ -1,0 +1,10 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "com.google.gson.**"
+    }
+  ]
+}

--- a/tests/src/com.google.code.gson/gson/2.3/user-code-filter.json
+++ b/tests/src/com.google.code.gson/gson/2.3/user-code-filter.json
@@ -5,6 +5,9 @@
     },
     {
       "includeClasses" : "com.google.gson.**"
+    },
+    {
+      "includeClasses" : "com_google_code_gson.gson.**"
     }
   ]
 }

